### ISSUE TITLE
Operand of null-aware operation '!' has type 'Locale' which excludes null.

### ISSUE THREAD

**Issue by alexi190** [medyas/flutter_zoom_drawer#47]
```
[+2159 ms] /C:/src/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_zoom_drawer-2.0.2+4/lib/flutter_zoom_drawer.dart:117:24:
Warning: Operand of null-aware operation '!' has type 'Locale' which excludes null.
[   +2 ms]  - 'Locale' is from 'dart:ui'.
[        ]       return ui.window.locale!.languageCode.toLowerCase();
[        ]                        ^
```

**Comment by enzo-desimone** [comment]
i have the same issue

**Comment by jemaineosia** [comment]
I am getting the same error.

**Comment by medyas** [comment]
this is just a warning, and removing the null safety operator will cause the app not to build. so just ignore it for now.

**Comment by thruthesky** [comment]
I got same warning.

**Comment by fikretsengul** [comment]
Yes it still exists on [2.0.2+4] even the changelog says "Fixed Locale warning"

**Comment by rymesaint** [comment]
the warning still exists

**Comment by YouSour** [comment]
for me i forgot this line : 
`await EasyLocalization.ensureInitialized(); 
`

**Comment by YouSour** [comment]
any news about this problem ? any one can fix it ?

**Comment by medyas** [comment]
@YouSour i don't think there is a fix right now for this.
it's just a warning and it won't effect the build process of the app

**Comment by YouSour** [comment]
@medyas thank you for information , i hope we will find a solution soon 😄

**Comment by Drolegc** [comment]
I will try to make a PullRequest "fixing" this warning.

**Comment by YouSour** [comment]
@Drolegc ok i wait for it :)

**Comment by fikretsengul** [comment]
@Drolegc Me too, thanks for your effort. :)

**Comment by nguyenlinhnttu** [comment]
the warning still exists, can't build app.

**Comment by maxTeste** [comment]
i'm having the error to, i think is similar 

lib/main.dart:265:104: Warning: Operand of null-aware operation '!' has type 'Locale' which excludes null.
 - 'Locale' is from 'dart:ui'.
            if (supportedLocale.languageCode == locale!.languageCode && supportedLocale.countryCode == locale!.countryCode) {

can someone help me plssss

**Comment by Drolegc** [comment]
> i'm having the error to, i think is similar
> 
> lib/main.dart:265:104: Warning: Operand of null-aware operation '!' has type 'Locale' which excludes null.
> 
> * 'Locale' is from 'dart:ui'.
>   if (supportedLocale.languageCode == locale!.languageCode && supportedLocale.countryCode == locale!.countryCode) {
> 
> can someone help me plssss

It's a warning, not an error. Can you build the app? Or this warning does not allow it?

**Comment by maxTeste** [comment]
@Drolegc yes i can build the app... but is a warning should be fixed
is there anything i can do to fix the warning?

**Comment by fikretsengul** [comment]
> @Drolegc yes i can build the app... but is a warning should be fixed
> is there anything i can do to fix the warning?

Yes it should be. Also I'm using the null safety of flutter and I cannot run my app without --no-sound-null-safety flag just because this issue.

**Comment by medyas** [comment]
just published a new version `2.1.0`, and removed `isRtl` method and by this the call to window.locale is removed, so the warnings will no longer be displayed.

**Comment by YouSour** [comment]
@medyas thank you :)
